pkg/subnet/public.go

```diff
@@ -151,3 +155,40 @@ func (d *PublicDeployer) createSubnetTx(controlKeys []string, threshold uint32,
 	opts := []common.Option{}
 	return wallet.P().IssueCreateSubnetTx(owners, opts...)
 }
+
+func (d *PublicDeployer) validateWalletIsSubnetOwner(controlKeys []string, threshold uint32) error {
+	var networkID uint32
+
+	switch d.network {
+	case models.Fuji:
+		networkID = avago_constants.FujiID
+	case models.Mainnet:
+		networkID = avago_constants.MainnetID
+	default:
+		return fmt.Errorf("unsupported public network")
+	}
+
+	sf, err := key.LoadSoft(networkID, d.privKeyPath)
```

Comment:
This should only be applied for `fuji` (on mainnet the subnet would have most probably had been created with ledger anyway)

Reply:
wallet is currently covering both fuji and mainnet, but should be fuji only. probably fix all of this for mainnet deploy.

Reply:
It's a bit odd though that just a couple of lines above we are supporting Mainnet. If you don't want to change this, then we should maybe remove the Mainnet case above.